src/app/forms/custom-sections/modified-weights/modified-weights.component.ts

```diff
@@ -0,0 +1,73 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { TechRecordGETHGV, TechRecordGETPSV, TechRecordGETTRL } from '@dvsa/cvs-type-definitions/types/v3/tech-record/tech-record-verb-vehicle-type';
+import { FormNode } from '@forms/services/dynamic-form.types';
+import { vehicleTemplateMap } from '@forms/utils/tech-record-constants';
+import { VehicleTypes } from '@models/vehicle-tech-record.model';
+import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+
+@Component({
+  selector: 'app-modified-weights',
+  templateUrl: './modified-weights.component.html',
+  styleUrls: ['./modified-weights.component.scss'],
+})
+export class ModifiedWeightsComponent implements OnInit {
+  @Input() vehicleType!: VehicleTypes;
+  @Input() changes!: Partial<TechRecordGETPSV | TechRecordGETHGV | TechRecordGETTRL>;
+
+  axleTemplate: FormNode[] | undefined;
+  psvGrossAxisChanged = false;
+  hgvGrossAxisChanged = false;
+  trlGrossAxisChanged = false;
+  hgvTrainAxisChanged = false;
+  psvTrainAxisChanged = false;
+  maxTrainAxisChanged = false;
+
+  constructor(private readonly technicalRecordService: TechnicalRecordService) {}
+
+  get isPSV(): boolean {
+    return this.vehicleType === VehicleTypes.PSV;
+  }
+
+  get isHGV(): boolean {
+    return this.vehicleType === VehicleTypes.HGV;
+  }
+
+  get isTRL(): boolean {
+    return this.vehicleType === VehicleTypes.TRL;
+  }
+
+  get psvChanges(): Partial<TechRecordGETPSV> | undefined {
+    return this.isPSV ? (this.changes as Partial<TechRecordGETPSV>) : undefined;
+  }
+
+  get hgvChanges(): Partial<TechRecordGETHGV> | undefined {
+    return this.isHGV ? (this.changes as Partial<TechRecordGETHGV>) : undefined;
+  }
+
+  get trlChanges(): Partial<TechRecordGETTRL> | undefined {
+    return this.isTRL ? (this.changes as Partial<TechRecordGETTRL>) : undefined;
+  }
+
+  get hgvAndTrlGrossAxisChanged() {
+    return this.hgvGrossAxisChanged || this.trlGrossAxisChanged;
+  }
+
+  ngOnInit(): void {
```

Comment:
Not without setting up an entire state for the techRecordChanges, which was the method of doing this ticket we decided against doing at the start, as it would have drastically changed our approach to figuring out the differences between the tech records, unless I'm misunderstanding you